libs/execution_context/ebpf_native.c

```diff
@@ -1398,3 +1451,44 @@ ebpf_native_get_count_of_maps(_In_ const GUID* module_id, _Out_ size_t* count_of
     ebpf_lock_unlock(&_ebpf_native_client_table_lock, state);
     EBPF_RETURN_RESULT(result);
 }
+
+static ebpf_result_t
+_ebpf_native_helper_address_changed(ebpf_program_t* program, void* context)
+{
+    ebpf_result_t return_value;
+    ebpf_native_helper_address_changed_context_t* helper_address_changed_context =
+        (ebpf_native_helper_address_changed_context_t*)context;
+
+    uint64_t* helper_function_addresses = NULL;
+    size_t helper_count = helper_address_changed_context->native_program->entry->helper_count;
+
+    if (helper_count == 0) {
+        return_value = EBPF_SUCCESS;
+        goto Done;
+    }
+
+    helper_function_addresses = ebpf_allocate(helper_count * sizeof(uint64_t));
+    if (helper_function_addresses == NULL) {
+        return_value = EBPF_NO_MEMORY;
+        goto Done;
+    }
+
+    return_value = ebpf_program_get_helper_function_addresses(program, helper_count, helper_function_addresses);
+
+    if (return_value != EBPF_SUCCESS) {
+        goto Done;
+    }
+
+    for (size_t i = 0; i < helper_count; i++) {
+        *(uint64_t*)&(helper_address_changed_context->native_program->entry->helpers[i].address) =
+            helper_function_addresses[i];
+    }
+
+    return_value = EBPF_SUCCESS;
+Done:
+    if (helper_function_addresses != NULL) {
```

Comment:
remove if test, to reduce number of code paths to test.  ebpf_free is safe to call with nullptr, like POSIX free()